src/coreneuron/io/nrn_filehandler.cpp

```diff
@@ -103,5 +105,13 @@ void FileHandler::read_checkpoint_assert() {
 
 void FileHandler::close() {
     F.close();
+
+    // Delete source file if read-mode is enabled and path matches '/dev/shm'
+    const auto& env = getenv("CORENEURON_SHM_DELETESOURCE");
+    if ((env == nullptr || std::string(env) != "OFF") && current_mode == std::ios::in &&
```

Comment:
The following would be `On`:
```
export CORENEURON_SHM_DELETESOURCE=0
```

Reply:
Yup, by default anything that is not `OFF`, it is considered `ON` by default.

Reply:
nitpick: could add `std::string(env) != "0"` to address the above confusion.

I would add a brief explanation/2-lines comment about the introduction of `CORENEURON_SHM_DELETESOURCE` env var here. It won't be obvious for non-BBP (users/)developers to understand why it's done like this.